default/scripting/species/common/happiness.macros

```diff
@@ -10,7 +10,18 @@ AVERAGE_HAPPINESS
             effects = SetHappiness value = Value +
                 min(abs(Value(Target.TargetHappiness) - Value), 1) *
                     (1 - 2*(Statistic If condition = And [Target (Value > Value(Target.TargetHappiness))]))
-            
+
+        EffectsGroup
+            scope = Source
+            activation = And [
+                Planet
+                (Source.LastTurnConquered < CurrentTurn - 5)
+            ]
+            effects = SetHappiness value = max(Value, 
```

Comment:
This seems a bit odd... why is it setting happiness directly and not target happiness? Will override some happiness modifiers and not others depending on effect priority, for all turns 5+ after an invasion...

Reply:
The idea is to make new settled (good environment) colonies to start with happiness 5. So the effect must work directly in the current value and not in the target.
But only on the turn in which the colony gets a (new) population!
How could that be done? How to reference the turn after a Colony Building or Colony Ship is stablished in the planet?

Reply:
@geoffthemedio As I understand it was done because with target happiness actual happiness reach 5 value only after 5 turns.

Reply:
Sound like you need an effect that sets the happiness just on the turn the planet is colonized, not every turn.

Reply:
I'm investigating this issue. My limited FOCS knowledge is a problem. Please check out if what if posted in [here](https://www.freeorion.org/forum/viewtopic.php?f=28&t=11604&p=100685#p100685) makes sense.